### PR TITLE
Orca openvino estimator

### PR DESCRIPTION
**Pull request by cyita**

### CODE REVIEW

File: pyzoo/zoo/orca/learn/openvino/estimator.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        self.node_num, self.core_num = get_node_and_core_number()
+        self.path = model_path
+        self.model = InferenceModel(supported_concurrent_num=self.core_num)
+        self.model.load_openvino(model_path=model_path,
+                                 weight_path=model_path[:model_path.rindex(".")] + ".bin")
+
+    def fit(self, data, epochs, **kwargs):
+        raise NotImplementedError
+
+    def predict(self, data, sc=None):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/openvino/estimator.py:66]: don't ask user to pass a `sc`; instead, get it from `rdd.ctx` or `init_nn_context`

File: pyzoo/zoo/orca/learn/openvino/estimator.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    def evaluate(self, data, **kwargs):
+        raise NotImplementedError
+
+    def get_model(self):
+        raise NotImplementedError
+
+    def save(self, checkpoint):
+        raise NotImplementedError
+
+    def load(self, checkpoint):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/openvino/estimator.py:127]: does it make sense to load from checkpoint?

File: pyzoo/zoo/orca/learn/openvino/estimator.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    def predict(self, data, sc=None):
+        def predict_transform(dict_data):
+            assert isinstance(dict_data, dict), "each shard should be an dict"
+            assert "x" in dict_data, "key x should in each shard"
+            return dict_data["x"]
+
+        if isinstance(data, SparkXShards):
+            assert sc is not None, "You should pass sc(spark context) if data is a XShards."
+            from zoo.orca.learn.tf.utils import convert_predict_to_xshard
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/openvino/estimator.py:74]: move this method to `zoo.orca.learn.utils`?

File: pyzoo/zoo/orca/learn/openvino/estimator.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            data = data.transform_shard(predict_transform)
+            result_rdd = self.model.distributed_predict(data.rdd, sc)
+            return convert_predict_to_xshard(result_rdd)
+        elif isinstance(data, (np.ndarray, list)):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/openvino/estimator.py:78]: we may refact this into a utility function, which can be used in other estimators too.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/PythonInferenceModel.scala [hunk 5]
```diff
@@ -18,7 +18,10 @@ package com.intel.analytics.zoo.pipeline.inference
 
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
 import com.intel.analytics.zoo.common.PythonZoo
-import java.util.{List => JList, ArrayList}
+import java.util.{ArrayList, List => JList}
+
+import org.apache.spark.api.java.JavaSparkContext
```
Comment (qiyuangong) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/PythonInferenceModel.scala:23]: This PR add Spark dependency into Python Inference model. Will this change impact inference model pojo sytle? @glorysdj

File: pyzoo/zoo/orca/learn/openvino/estimator.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+            if sc is None:
+                return self.model.predict(data)
```
Comment (qiyuangong) [inline pyzoo/zoo/orca/learn/openvino/estimator.py:80]: Need to ensure batch size of data is equal to model batch size.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/PythonInferenceModel.scala (hunk 5 above)
Comment (jason-dai) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/PythonInferenceModel.scala:23]: We can move the distributed prediction to orca instead

File: pyzoo/test/zoo/orca/learn/spark/test_estimator_openvino.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+from zoo.orca.learn.openvino.estimator import Estimator
+from bigdl.dataset.base import maybe_download
+
+resource_path = os.path.join(os.path.split(__file__)[0], "../../../resources")
+
+
+class TestEstimatorForOpenVINO(TestCase):
+    def test_openvino(self):
+        with tempfile.TemporaryDirectory() as local_path:
+            model_url = "http://10.239.45.10:8081/repository/raw/analytics-zoo-data/" \
```
Comment (qiyuangong) [inline pyzoo/test/zoo/orca/learn/spark/test_estimator_openvino.py:33]: Replace local IP with replaceable variable, https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/test/zoo/pipeline/inference/test_inference_model.py#L33

**Comment by cyita** [comment]
Jenkins passed http://10.239.47.210:18888/job/ZOO-PR-Validation/4526/

**Review by qiyuangong — APPROVED** [review]
LGTM

**Pull request merged**